### PR TITLE
Inline useless wrapper functions

### PR DESCRIPTION
**Pull request by rudolfbyker**
- Inline postForm, postJSON and getJSON.
- Make 'createSchemaField' a method of Client.

This gets us halfway to where we want to be with refactoring. But I figured it's a good stepping stone to first merge this.

The inlining looks like it might increase code duplication, but the goal is to reduce it much more, and in a smarter way, in the next PR.

### CODE REVIEW

File: lib/solr.js [hunk 1]
```diff
@@ -827,120 +884,60 @@ Client.prototype.ping = function (callback) {
...
     method: 'POST',
-    headers: headers,
-    path: params.fullPath,
-    family: params.ipVersion,
+    headers: {
+      'content-type': 'application/json; charset=utf-8',
+      'content-length': Buffer.byteLength(json),
+      accept: 'application/json; charset=utf-8',
+    },
+    path: 'http://localhost:8983/solr/testcore/schema',
```
Comment (kibertoad) [inline lib/solr.js:917]: why does this have hardcoded path?

**Comment by kibertoad** [comment]
Can you explain what you are trying to do with this refactoring? I don't think I understand what it does and why.

File: lib/solr.js (hunk 1 above)
Comment (rudolfbyker) [reply lib/solr.js:917]: It's part of the original code. I just moved the method, I did not write it. Currently this method is only used during tests, so the URL is always the same. We should fix that in a separate issue.

**Comment by rudolfbyker** [comment]
> Can you explain what you are trying to do with this refactoring? I don't think I understand what it does and why.

Exactly as described here https://github.com/lbdremy/solr-node-client/pull/255 . I'm reducing the duplication of _logic_ (not necessarily just lines of code) to make it more maintainable.

This is just a stepping stone towards that, by inlining useless wrapper functions, to allow us to extract the logic which is actually shared in a smarter way (In the next PR, because you requested that we reduce the scope and take smaller steps).

File: lib/solr.js (hunk 1 above)
Comment (rudolfbyker) [reply lib/solr.js:917]: Actually I can include that fix in this PR. Just a moment... :)

**Comment by rudolfbyker** [comment]
I you need to see the bigger picture, I can put everything in one PR, and then you would be able to see the clean result I'm aiming for.

File: lib/solr.js [hunk 2]
```diff
@@ -550,19 +549,41 @@ Client.prototype.update = function (data, options, callback) {
...
-    ipVersion: this.options.ipVersion,
-    request: this.request,
+    method: 'POST',
+    headers: {
+      'content-type': 'application/json; charset=utf-8',
+      'content-length': Buffer.byteLength(json),
+      accept: 'application/json; charset=utf-8',
+    },
+    path: fullPath,
+    family: this.options.ipversion,
```
Comment (kibertoad) [inline lib/solr.js:562]: still trying to wrap my head around all these changes. So I see a new field `family` that wasn't here before, and I don't see some of the fields that previously were here. Do all of them come from `this.options.request` now?
sorry for the long delay on reviewing this PR, I have a rather uneasy feeling about it due to the rather opaque changes in it. If you have some time in the nearby future, can we talk them through and push forward with this PR?

Comment (kibertoad) [reply lib/solr.js:562]: @rudolfbyker Sorry, it took me a while and adding TypeScript to fully understand what was going here, and I envy your capability to comprehend rather messy code here and actually fix what was broken before. I believe that your change is completely correct here and it should be done like this. Unfortunately, my TypeScript addition and class refactorings caused conflicts in your PR, but it should be easier to make those changes now that we have the safety net.
Would you be open to rebasing this PR? Alternatively I can use it as a basis for a reimplementation for the same refactorings that you were going for.

Comment (rudolfbyker) [reply lib/solr.js:562]: Ah, I'm glad we're on the same page now :) I'm definitely open to rebasing this, or throwing it away completely and using just the spirit of it to help future work. I saw the many other PRs and commits, and will try to make time ASAP to look at those to get up to speed with what you are doing.

Comment (kibertoad) [reply lib/solr.js:562]: Appreciate it dearly!
We've been mostly adding types, I did my best to avoid touching request-related code for now.

**Comment by rudolfbyker** [comment]
I'll open separate PRs for this, and try to do it in smaller steps this time. The problem with smaller steps when refactoring messy code, however, is that the intermediate steps are also far from ideal. Sometimes you have to make something slightly worse to make it much better. Like when cleaning your house.

**Pull request closed without merging**